### ISSUE TITLE
Using CismetExecutors in CismetThreadPool

### ISSUE THREAD

**Issue by dameiers** [cismet/cismet-commons#31]
The deprecated `CismetThreadPool` class uses a `java.util.concurrent.ThreadPoolExecutor` to start threads. This Executor implementation internally uses a Future which swallows  all uncaught exceptions until calling get() on the respective Future. Since this get call is not mandatory (and not possible when usig execute) uncaught exception are not recognized by the DefaultExceptionHandler and do not appear in the logging mechanism. See [source 1](http://cs.oswego.edu/pipermail/concurrency-interest/2011-April/007866.html) and [source 2](https://www.securecoding.cert.org/confluence/display/java/TPS03-J.+Ensure+that+tasks+executing+in+a+thread+pool+do+not+fail+silently) for more informtation.

The appropriate solution is to use the ExecutorServices provided by `CismetConcurrency` and `CismetExecutors` classes, which both guarantee that uncaught exceptions are carried over to the UncaughtExceptionHandler.

Using the `CismetExecutors` in the deprecated `CismetThreadPool` class provides  an intermediate solution for all the legacy code that still use `CismetThreadPool` and is not refactored to the above outlined appropriate solution in order to ensure logging by the DefaultExceptionHandler.
## tl;dr
- **Never** use  `CismetThreadPool`
- Use **CismetConcurrency** or **CismetExecutors** classes instead
